rust/signed_doc/src/metadata/doc_type.rs

```diff
@@ -0,0 +1,225 @@
+//! Document Type.
+
+use std::fmt::{Display, Formatter};
+
+use catalyst_types::{
+    problem_report::ProblemReport,
+    uuid::{CborContext, Uuid, UuidV4},
+};
+use minicbor::{Decode, Decoder, Encode};
+use tracing::warn;
+
+use crate::decode_context::{ConversionPolicy, DecodeContext};
+
+/// List of `UUIDv4` document type.
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct DocType(Vec<UuidV4>);
+
+impl DocType {
+    /// Get a list of `UUIDv4` document types.
+    #[allow(dead_code)]
```

Comment:
Why does this need dead_code.
If its public and accessible outside the crate it should not require it?